packages/backend/src/controllers/projectController.ts

```diff
@@ -505,4 +513,109 @@ export class ProjectController extends BaseController {
             results: undefined,
         };
     }
+
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Get('{projectUuid}/dashboards')
+    @OperationId('getDashboards')
+    async getDashboards(
+        @Path() projectUuid: string,
+        @Request() req: express.Request,
+    ): Promise<ApiGetDashboardsResponse> {
+        this.setStatus(200);
+
+        const chartUuid: string | undefined =
+            typeof req.query.chartUuid === 'string'
+                ? req.query.chartUuid.toString()
+                : undefined;
+
+        const includePrivate = req.query.includePrivate !== 'false';
+
+        const results = await this.services
+            .getDashboardService()
+            .getAllByProject(req.user!, projectUuid, chartUuid, includePrivate);
+
+        return {
+            status: 'ok',
+            results,
+        };
+    }
+
+    @Middlewares([
+        allowApiKeyAuthentication,
+        isAuthenticated,
+        unauthorisedInDemo,
+    ])
+    @SuccessResponse('201', 'Created')
+    @Post('{projectUuid}/dashboards')
+    @OperationId('createDashboard')
+    async createDashboard(
+        @Path() projectUuid: string,
+        @Body() body: DuplicateDashboardParams | CreateDashboard,
+        @Request() req: express.Request,
+        @Query() duplicateFrom?: string,
+    ): Promise<ApiCreateDashboardResponse> {
+        const dashboardService = this.services.getDashboardService();
+        this.setStatus(201);
+
+        let results: ApiCreateDashboardResponse['results'];
+
+        if (duplicateFrom) {
```

Comment:
This is ok because you copied it from the existing browser, but we should move this logic to the service IMO. We can do it on a separate PR or make a ticket.